### PR TITLE
add htmgo.dev

### PR DESCRIPTION
**Pull request by maddalax**
## We want to ensure high quality of the packages. Make sure that you've checked the boxes below before sending a pull request.

- [x] I have read the [Contribution Guidelines](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#contribution-guidelines)
- [x] I have read the [Maintainers Note](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#maintainers)
- [x] I have read the [Quality Standards](https://github.com/avelino/awesome-go/blob/main/CONTRIBUTING.md#quality-standards)

_Not every repository (project) will require every option, but most projects should. Check the Contribution Guidelines for details._

- [x] The repo documentation has a pkg.go.dev link.
- [x] The repo documentation has a coverage service link.
- [x] The repo documentation has a goreportcard link.
- [x] The repo has a version-numbered release and a go.mod file.
- [x] The repo has a continuous integration process that automatically runs tests that must pass before new pull requests are merged.
- [x] Continuous integration is used to attempt to catch issues prior to releasing this package to end-users.

## Please provide some links to your package to ease the review

- [x] https://github.com/maddalax/htmgo
- [x] pkg.go.dev: docs are on https://htmgo.dev/docs but I'll include https://pkg.go.dev/github.com/maddalax/htmgo/framework@v1.0.2/h
- [x] goreportcard.com: https://goreportcard.com/report/github.com/maddalax/htmgo
- [x] coverage service link: https://app.codecov.io/github/maddalax/htmgo

## Pull Request content

- [x] The package has been added to the list in alphabetical order.
- [x] The package has an appropriate description with correct grammar.
- [x] As far as I know, the package has not been listed here before.

## Category quality

_Note that new categories can be added only when there are 3 packages or more._

Packages added a long time ago might not meet the current guidelines anymore. It would be very helpful if you could check 3-5 packages above and below your submission to ensure that they also still meet the Quality Standards.

Please delete one of the following lines:

-[x] I removed the following packages around my addition: (please give a short reason for each removal)
**gospin** - not updated in over 5 years, very little usage

Thanks for your PR, you're awesome! :sunglasses:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2408,9 +2408,9 @@ _Libraries and tools for templating and lexing._
 - [extemplate](https://github.com/dannyvankooten/extemplate) - Tiny wrapper around html/template to allow for easy file-based template inheritance.
 - [fasttemplate](https://github.com/valyala/fasttemplate) - Simple and fast template engine. Substitutes template placeholders up to 10x faster than [text/template](https://golang.org/pkg/text/template/).
 - [gomponents](https://www.gomponents.com) - HTML 5 components in pure Go, that look something like this: `func(name string) g.Node { return Div(Class("headline"), g.Textf("Hi %v!", name)) }`.
-- [gospin](https://github.com/m1/gospin) - Article spinning and spintax/spinning syntax engine, useful for A/B, testing pieces of text/articles and creating more natural conversations.
```
Comment (yassinebenaid) [inline README.md:2411]: Can you provide the reason for removing this project !!!.

Comment (maddalax) [reply README.md:2411]: @yassinebenaid I did in the README!

```
-[x] I removed the following packages around my addition: (please give a short reason for each removal)
gospin - not updated in over 5 years, very little usage
```

Comment (yassinebenaid) [reply README.md:2411]: Ohhh  sorry, didn't notice. Thanks.

**Review by yassinebenaid — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (maddalax) [reply README.md:2411]: no worries, thanks for merging!

**Pull request merged**